### ISSUE TITLE
Compatibility with graylog 2.3

### ISSUE THREAD

**Issue by ricoli** [graylog-labs/graylog-plugin-aws#41]
So I tried installing this plugin in the first beta of graylog 2.3. The plugin installs fine and shows in the list of plugins, but there's no configuration section showing up for the plugin. Is support for 2.3 in the pipeline?

**Comment by bernd** [comment]
@ricoli Thanks for the report. We will make sure to release a version that is compatible with Graylog 2.3. I will let you know once this is done.

**Comment by bernd** [comment]
@ricoli I just released version 2.3.0-rc.1 which is compatible with Graylog 2.3. It would be great if you could test this and report back any issues. Thank you!

https://github.com/graylog-labs/graylog-plugin-aws/releases/tag/2.3.0-rc.1

**Comment by ricoli** [comment]
Hi @bernd and thank you for releasing the new version. Unfortunately I do get a lot of errors when graylog v2.3.0-7 starts with the aws plugin:
```
2017-07-27T13:20:45.369+01:00 ERROR [CmdLineTool] Guice error (more detail on log level debug): Could not find a suitable constructor in org.graylog.aws.inputs.cloudtrail.CloudTrailCodec. Classes must have either one (and only one) constructor annotated with @Inject or a zero-argument constructor that is not private.
2017-07-27T13:20:45.369+01:00 ERROR [CmdLineTool] Guice error (more detail on log level debug): Could not find a suitable constructor in org.graylog.aws.inputs.cloudtrail.CloudTrailInput. Classes must have either one (and only one) constructor annotated with @Inject or a zero-argument constructor that is not private.
2017-07-27T13:20:45.369+01:00 ERROR [CmdLineTool] Guice error (more detail on log level debug): Could not find a suitable constructor in org.graylog.aws.inputs.cloudtrail.CloudTrailInput$Config. Classes must have either one (and only one) constructor annotated with @Inject or a zero-argument constructor that is not private.
2017-07-27T13:20:45.369+01:00 ERROR [CmdLineTool] Guice error (more detail on log level debug): Could not find a suitable constructor in org.graylog.aws.inputs.cloudtrail.CloudTrailTransport. Classes must have either one (and only one) constructor annotated with @Inject or a zero-argument constructor that is not private.
2017-07-27T13:20:45.370+01:00 ERROR [CmdLineTool] Guice error (more detail on log level debug): Could not find a suitable constructor in org.graylog.aws.inputs.flowlogs.FlowLogCodec. Classes must have either one (and only one) constructor annotated with @Inject or a zero-argument constructor that is not private.
2017-07-27T13:20:45.370+01:00 ERROR [CmdLineTool] Guice error (more detail on log level debug): Could not find a suitable constructor in org.graylog.aws.inputs.flowlogs.FlowLogTransport. Classes must have either one (and only one) constructor annotated with @Inject or a zero-argument constructor that is not private.
2017-07-27T13:20:45.370+01:00 ERROR [CmdLineTool] Guice error (more detail on log level debug): Could not find a suitable constructor in org.graylog.aws.inputs.flowlogs.FlowLogsInput. Classes must have either one (and only one) constructor annotated with @Inject or a zero-argument constructor that is not private.
2017-07-27T13:20:45.370+01:00 ERROR [CmdLineTool] Guice error (more detail on log level debug): Could not find a suitable constructor in org.graylog.aws.inputs.flowlogs.FlowLogsInput$Config. Classes must have either one (and only one) constructor annotated with @Inject or a zero-argument constructor that is not private.
2017-07-27T13:20:45.370+01:00 ERROR [CmdLineTool] Guice error (more detail on log level debug): No implementation for org.graylog.aws.inputs.cloudtrail.CloudTrailCodec$Factory was bound.
2017-07-27T13:20:45.370+01:00 ERROR [CmdLineTool] Guice error (more detail on log level debug): No implementation for org.graylog.aws.inputs.flowlogs.FlowLogCodec$Factory was bound.
2017-07-27T13:20:45.370+01:00 ERROR [CmdLineTool] Guice error (more detail on log level debug): No implementation for org.graylog.aws.inputs.cloudtrail.CloudTrailInput$Factory was bound.
2017-07-27T13:20:45.371+01:00 ERROR [CmdLineTool] Guice error (more detail on log level debug): No implementation for org.graylog.aws.inputs.flowlogs.FlowLogsInput$Factory was bound.
2017-07-27T13:20:45.371+01:00 ERROR [CmdLineTool] Guice error (more detail on log level debug): No implementation for org.graylog.aws.inputs.cloudtrail.CloudTrailTransport$Factory was bound.
2017-07-27T13:20:45.371+01:00 ERROR [CmdLineTool] Guice error (more detail on log level debug): No implementation for org.graylog.aws.inputs.flowlogs.FlowLogTransport$Factory was bound.
```
Graylog starts just fine if I don't have the aws plugin installed. I've used the manual installation procedure (copying the jar file).

**Comment by bernd** [comment]
@ricoli Thank you for the feedback. I will investigate!

**Comment by bernd** [comment]
@ricoli I found and fixed a few issues. Next try: https://github.com/graylog-labs/graylog-plugin-aws/releases/tag/2.3.0-rc.2 :smiley:

**Comment by ricoli** [comment]
Hi @bernd and thank you so much for fixing those issues! I can confirm that the plugin works now for the functionality I'm using it for (cloudtrail logs), can't speak for the rest though. Wonderful work 👍

**Comment by bernd** [comment]
@ricoli Great, thank you very much for testing!

I just released version 2.3.0 of the plugin. https://github.com/graylog-labs/graylog-plugin-aws/releases/tag/2.3.0
